File: LastOcc_LinkedList/Find.ts
class NODE 
{
    data : number;
    next : NODE | null;

    constructor()
    {
        this.data = 0;
        this.next = null;
    }
}

class SinglyLL
{
    Head : NODE | null;

    constructor()
    {
        this.Head = null;
    }

    InsertFirst(value : number)
    {
        var newn : NODE = new NODE();
        newn.data = value;
        newn.next = null;
        
        if(this.Head == null)
        {
            this.Head = newn;
        }
        else
        {
            newn.next = this.Head;
            this.Head = newn;
        }
    }

    InsertLast(value : number)
    {
        var temp : NODE |null = this.Head;

        var newn : NODE = new NODE();
        newn.data = value;
        newn.next = null;

        if(temp == null)
        {
            temp = newn;
        }
        else
        {
            while(temp.next != null)
            {
                temp = temp.next;
            }
            temp.next = newn;
        }
    }

    LastOcc(value : number) : number 
    {
        var icnt : number = 1;
        var ipos : number = 0;
        var temp : NODE | null = this.Head;

        while(temp != null)
        {
            if(temp.data == value)
            {
                ipos = icnt;
            }
            icnt++;
            temp = temp.next;
        }

        return ipos;
    }
    Count() : number
    {
        var icnt : number = 0;
        var temp : NODE | null = this.Head;

        while(temp != null)
        {
            icnt++;
            temp = temp.next;
        }

        return icnt;
    }

    Display()
    {
        var temp : NODE | null = this.Head;

        while(temp != null)
        {
            process.stdout.write("|"+temp.data+"|->");
            temp = temp.next;
        }
    }
}

function main()
{
    console.log("Jay Ganesh....");

    var sobj : SinglyLL = new SinglyLL();
    var iret : number = 0;

    sobj.InsertFirst(51);
    sobj.InsertFirst(11);
    sobj.InsertFirst(10);
    sobj.InsertFirst(5);
    sobj.InsertLast(101);
    sobj.InsertLast(201);
    sobj.InsertLast(11);
    sobj.InsertLast(301);
    sobj.Display();

    iret = sobj.Count();
    console.log("\nCount of Elements Present in Linked is : %d",iret);

    iret = sobj.LastOcc(11);
    console.log("Last Occurance of Entered Element is At Position number : %d",iret);
}

main();

File: LastOcc_LinkedList/Find.go
package main
import "fmt"

type Node struct {
	data int;
	next * Node;
}

type NODE = Node;
type PNODE = * Node;
type PPNODE = ** Node;

func InsertFirst(Head PPNODE,value int) {
	var newn = new(NODE);
	newn.data = value;
	newn.next = nil;


	if *Head == nil {
		*Head = newn;
	} else {
		newn.next = *Head;
		*Head = newn;
	}
}

func InsertLast(Head PPNODE,value int) {
	var newn = new(NODE);
	var temp = *Head;
	newn.data = value;
	newn.next = nil;

	if(temp == nil) {
		temp = newn;
	} else {
		for(temp.next != nil) {
			temp = temp.next;
		}
		temp.next = newn;
	}
}

func Count(Head PNODE) int {
	var icnt int = 0;

	for(Head != nil) {
		icnt++;
		Head = Head.next;
	}

	return icnt;
}

func LastOcc(Head PNODE,value int) int {
	var icnt = 1;
	var ipos = 0;

	for(Head != nil) {
		if(Head.data == value) {
			ipos = icnt;
		} 
		icnt++;
		Head = Head.next;
	}

	return ipos;
}

func Display(Head PNODE) {

	for Head != nil {
		fmt.Printf("|%d|->",Head.data);
		Head = Head.next;
	}
}

func main() {
	var First PNODE = nil;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");

	InsertFirst(&First,51);
	InsertFirst(&First,11);
	InsertFirst(&First,10);
	InsertFirst(&First,5);
	InsertLast(&First,101);
	InsertLast(&First,11);
	InsertLast(&First,201);
	
	Display(First);

	iret = Count(First);
	fmt.Printf("\nCount of Elements from Linked List is : %d",iret);

	iret = LastOcc(First,11);
	fmt.Printf("\nLast Occurance of Element is at Position Number : %d",iret);
}